constructor/construct.py

```diff
@@ -1079,68 +133,49 @@ def parse(path, platform):
     return res
 
 
-def verify(info):
-    types_key = {}  # maps key to types
-    required_keys = set()
-    obsolete_keys = set()
-    for key, required, types, descr in KEYS:
-        types_key[key] = types
-        if required:
-            required_keys.add(key)
-        if "Obsolete" in descr:
-            obsolete_keys.add(key)
+# this is actually not an error, therefore the naming is okay
+class DeprecatedFieldWarning(ValidationError):
+    pass
 
-    for key in info:
-        if key not in types_key:
-            sys.exit("Error: unknown key '%s' in construct.yaml" % key)
-        elt = info[key]
-        if key in obsolete_keys:
-            logger.warning("key '%s' is obsolete. Its value '%s' is being ignored.", key, elt)
-        types = types_key[key]
-        if not isinstance(elt, types):
-            sys.exit("Error: key '%s' points to %s,\n       expected %s" % (key, type(elt), types))
 
-    for key in required_keys:
-        if key not in info:
-            sys.exit("Error: Required key '%s' not found in construct.yaml" % key)
+def deprecated_validator(validator, value, instance, schema):
+    if value and instance is not None:
+        print(value)
+        print(instance)
+        print(schema)
+        yield DeprecatedFieldWarning(f"'{schema['title']}' is deprecated.\n{schema['description']}")
 
-    pat = re.compile(r"[\w][\w\-\.]*$")
-    for key in "name", "version":
-        value = info[key]
-        if not pat.match(value) or value.endswith((".", "-")):
```

Comment:
This is the only part that I haven't figured out yet. The regex I am proposing can't match "endswith" for now. I'll give it a try.

Reply:
Solved